cmd/tempo/app/modules.go

```diff
@@ -156,24 +158,27 @@ func (t *App) initQuerier() (services.Service, error) {
 
 func (t *App) initQueryFrontend() (services.Service, error) {
 	var err error
-	t.frontend, err = cortex_frontend.New(t.cfg.Frontend.Config, util.Logger, prometheus.DefaultRegisterer)
+
+	originalTripper, v1, _, err := cortex_frontend.InitFrontend(t.cfg.Frontend.Config, frontend.CortexNoQuerierLimits{}, 0, util.Logger, prometheus.DefaultRegisterer)
 	if err != nil {
 		return nil, err
 	}
+	t.frontend = v1
 
 	// custom tripperware that splits requests
-	tripperware, err := frontend.NewTripperware(t.cfg.Frontend, util.Logger, prometheus.DefaultRegisterer)
+	shardingTripper, err := frontend.NewTripperware(t.cfg.Frontend, util.Logger, prometheus.DefaultRegisterer)
 	if err != nil {
 		return nil, err
 	}
-	// tripperware will be called before f.roundTripper (which calls roundtripgrpc)
-	t.frontend.Wrap(tripperware)
+	finalTripper := shardingTripper(originalTripper)
+
+	handler := cortex_transport.NewHandler(t.cfg.Frontend.Config.Handler, finalTripper, util.Logger, prometheus.DefaultRegisterer)
```

Comment:
I think we may have the same issue of broken instrumentation as we did with the querier? Missing instrumentation since we're using the handler directly.. but we can fix that in a follow up PR.